### PR TITLE
WIP: implement IBigtableDataClient interface

### PR DESCRIPTION
**Pull request by elisheva-qlogic**
`IBigtableDataClient` interface implemented
`mutateRow` and `checkAndMutateRowAsync` methods implemented

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java [hunk 1]
```diff
@@ -80,7 +92,7 @@
  * @author sduskis
  * @version $Id: $Id
  */
-public class BigtableDataGrpcClient implements BigtableDataClient {
+public class BigtableDataGrpcClient implements BigtableDataClient, IBigtableDataClient {
```
Comment (sduskis) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: Please create a separate class that implements `IBigtableDataClient` and uses a `BigtableDataClient`

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/HBaseRequestAdapter.java [hunk 2]
```diff
@@ -222,12 +223,13 @@ public ReadModifyWriteRowRequest adapt(Increment increment) {
    * <p>adapt.</p>
    *
    * @param put a {@link org.apache.hadoop.hbase.client.Put} object.
-   * @return a {@link com.google.bigtable.v2.MutateRowRequest} object.
+   * @return a {@link RowMutation} object.
    */
-  public MutateRowRequest adapt(Put put) {
-    RowMutation rowMutation = newRowMutationModel(put.getRow());
+  public RowMutation adapt(Put put) {
```
Comment (sduskis) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/HBaseRequestAdapter.java]: Please make this change in a separate PR before this one.

**Comment by sduskis** [comment]
@elisheva-qlogic can we close this as well?

**Comment by elisheva-qlogic** [comment]
yes
